src/mongo_client_options.ts

```diff
@@ -0,0 +1,143 @@
+import type { MongoCredentials } from './cmap/auth/mongo_credentials';
+import type { CompressorName } from './cmap/wire_protocol/compression';
+import type { AutoEncryptionOptions } from './deps';
+import type { Logger } from './logger';
+import type { DriverInfo, MongoClientOptions, PkFactory } from './mongo_client';
+import type { ReadConcern } from './read_concern';
+import type { ReadPreference } from './read_preference';
+import type { WriteConcern } from './write_concern';
+
+import type { ConnectionOptions as TLSConnectionOptions } from 'tls';
+import type { TcpSocketConnectOpts as ConnectionOptions } from 'net';
+import type { BSONSerializeOptions } from './bson';
+import { MongoParseError } from './error';
+import { URL } from 'url';
+
+/**
+ * Mongo Client Options
+ * @internal
+ */
+export interface MongoOptions
```

Comment:
After our conversation yesterday, I'm wondering if `MongoOptions` shouldn't just be a "resolved" version of `MongoClientOptions`? Should `MongoOptions` basically be:
```
export type MongoOptions = Omit<MongoClientOptions, ....> & {
  moreSpecificOptions: any;
  newOptionNames: any
}
```

I wonder if that also means that you propagate the doc strings for driver dev use?

Reply:
Good idea! 👍  It does bring along the docs which is very nice. And in the case of tls I can redo that one prop in MongoOptions to keep the table I made as internal docs.